tutorials/glod-game/steps/step-3.md

```diff
@@ -999,408 +1000,409 @@ To do this we will:
+    (When :Predicate (IsMoreEqual .walking-image-index-max)
+          :Action (-> 0 > .walking-image-index))
+    (Pause .walking-animation-speed))
+
+  ;; ------- Button Inputs ----------
+  (defshards button-inputs []
+    (Inputs.KeyDown
+    :Key "left"
+    :Action (->
+              (Msg "left")
+              1 > .character-state
+              0 > .character-direction))
+
+    (Inputs.KeyDown
+    :Key "right"
+    :Action (->
+              (Msg "right")
+              2 > .character-state
+              1 > .character-direction))
+
+    (Inputs.KeyDown
+    :Key "up"
+    :Action (->
+              (Msg "up")
+              3 > .character-state))
+
+    (Inputs.KeyUp
+    :Key "left"
+    :Action (->
+              0 > .character-state))
+
+    (Inputs.KeyUp
+    :Key "right"
+    :Action (->
+              0 > .character-state)))
+
+  ;;---------- main-wire ------------
+  (defloop main-wire
+    (Setup
+    (initialize-character))
+
+    (Step idle-animation)
+    (Step walking-animation)
+
+    (GFX.MainWindow
+    :Title "MainWindow" :Width 1920 :Height 1080
+    :Contents
+    (-> (Setup
+          (GFX.DrawQueue) >= .ui-draw-queue
+          (GFX.UIPass .ui-draw-queue) >> .render-steps)
+        .ui-draw-queue (GFX.ClearQueue)
 
-    (defmesh main)
-    (schedule main main-wire)
-    (run main (/ 1.0 60))
-    ```
+        (UI
+          .ui-draw-queue
+          (->
+          (UI.Area :Position (float2 0 0)
+                    :Anchor Anchor.Center
+                    :Contents
+                    (->
+                    .character-state
+                    (Match [0 (-> .character-direction
+                                  (Match [0 (-> load-texture .idle-left-image-array (Take .idle-image-index) (UI.Image :Scale (float2 0.2)))
+                                          1 (-> load-texture .idle-right-image-array (Take .idle-image-index) (UI.Image :Scale (float2 0.2)))]
+                                          :Passthrough false))
+                            1 (-> load-texture .walking-left-image-array (Take .walking-image-index) (UI.Image :Scale (float2 0.2)))
+                            2 (-> load-texture .walking-right-image-array (Take .walking-image-index) (UI.Image :Scale (float2 0.2)))
+                            3 (-> load-texture .character-jumping (UI.Image :Scale (float2 0.2)))]
+                            :Passthrough false)))))
+
+        (button-inputs)
+
+        (GFX.Render :Steps .render-steps))))
+
+
+  (defmesh main)
+  (schedule main main-wire)
+  (run main (/ 1.0 60))
+  ```
 
 When you run the programme, Glod will  return to his idle animation whenever you stop pressing the left, right or up buttons and he should be facing the correct direction. Now we have to do the same thing for our jump animation
 
 - Download Jumping Image [here](https://drive.google.com/drive/folders/1l5y2PoZARAukfsNz_qeQXQblrrpnwy6i?usp=share_link).
 
 === "Code Added"
     
-    ```{.clojure .annotate linenums="1"}
```

Comment:
You might want to double check these code blocks, I think the indentation was required this way to correctly show the content tabs (https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#feature-enabled)

You can compare the live docs (https://docs.fragnova.com/learn/shards/tutorials/glod-game/steps/step-3/) with one you build locally.
You can check the shards repo's `.github\workflows\build-doc.yml` for steps required to build the docs; the main requirements are: copy or symlink the tutorials folder to the correct locations and run `mkdocs build`

Reply:
You're right, my mistake, I was testing the docs with the VSCode preview which isn't accurate. The latest commit [e1f3e72](https://github.com/fragcolor-xyz/shards-examples/pull/67/commits/e1f3e7242832cb0fae5a88b91ad51b076aec2a78) has reverted some changes regarding the indentation and a few other misc fixes were also made while comparing the live build against a local build of the docs using mkdocs. The code blocks should all appear properly now, unless I missed any. 

"Full Code So Far" blocks are now all properly nested as the 2nd tab while the 1st tab is always the "Code Added" tab so the page will not appear to be too long so it is more comfortable to scroll through while still able to access the full code at the end of each step.

I think `.github\workflows\build-doc.yml` might be missing though

Reply:
It should be in the shards repository, not the examples one.